lib/frr_pthread.c

```diff
@@ -126,16 +130,31 @@ void frr_pthread_destroy(struct frr_pthread *fpt)
 {
 	thread_master_free(fpt->master);
 
+	pthread_mutex_destroy(&fpt->mtx);
 	pthread_mutex_destroy(fpt->running_cond_mtx);
 	pthread_cond_destroy(fpt->running_cond);
+	if (fpt->name)
+		XFREE(MTYPE_FRR_PTHREAD, fpt->name);
 	XFREE(MTYPE_PTHREAD_PRIM, fpt->running_cond_mtx);
 	XFREE(MTYPE_PTHREAD_PRIM, fpt->running_cond);
 	XFREE(MTYPE_FRR_PTHREAD, fpt);
 }
 
+void frr_pthread_set_name(struct frr_pthread *fpt, const char *name)
+{
+	pthread_mutex_lock(&fpt->mtx);
+	{
+		if (fpt->name)
+			XFREE(MTYPE_FRR_PTHREAD, fpt->name);
+		fpt->name = XSTRDUP(MTYPE_FRR_PTHREAD, name);
+	}
+	pthread_mutex_unlock(&fpt->mtx);
+	thread_master_set_name(fpt->master, name);
+}
+
 struct frr_pthread *frr_pthread_get(unsigned int id)
```

Comment:
nit: change from `unsigned int` to `uint32_t`?

Reply:
yup